File: src/main/java/com/blink/backend/controller/message/dto/WhatsAppStatusDto.java
package com.blink.backend.controller.message.dto;

import com.blink.backend.domain.model.message.WhatsAppStatus;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Builder;
import lombok.Getter;

import static com.blink.backend.domain.model.message.WhatsAppStatus.CONNECTED;
import static com.blink.backend.domain.model.message.WhatsAppStatus.SHUTDOWN;

@Getter
@Builder
public class WhatsAppStatusDto {
    private WhatsAppStatus status;
    private String connectedPhoneNumber;

    @JsonIgnore
    public boolean isShutdown() {
        return SHUTDOWN.equals(status);
    }

    @JsonIgnore
    public boolean isConnected() {
        return CONNECTED.equals(status);
    }

    @JsonIgnore
    public boolean isNotConnected() {
        return !isConnected();
    }
}

File: src/main/java/com/blink/backend/controller/chat/dto/WhatsAppStatusDto.kt
package com.blink.backend.controller.chat.dto

import com.blink.backend.domain.clinic.chat.model.WhatsAppStatus

data class WhatsAppStatusDto(val status: String, val connectedNumber: String?) {
    companion object {
        fun fromDomain(domain: WhatsAppStatus): WhatsAppStatusDto {
            return WhatsAppStatusDto(status = domain.status.name, connectedNumber = domain.whatsAppNumber)
        }
    }
}